kafka/producer.go

```diff
@@ -251,10 +264,13 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 			ctx = queuecontext.DetachedContext(ctx)
 		}
 		p.client.Produce(ctx, record, func(msg *kgo.Record, err error) {
-			defer func() {
-				wg.Done()
-				span.End()
-			}()
+			_, span := p.tracer.Start(ctx, "producer.Produce", trace.WithAttributes(
```

Comment:
This doesn't seem right to me, the span is called after the event has been produced to Kafka, so what are we tracking here?

What could be an acceptable approach would be to start the span in line `236`, so we have a span tracking the time it takes to create and produce a `kgo.Record` individually, and also can set the correct state, and track the production errors if any. That's why I was implying that the kgo hooks won't add much value then.

Reply:
And as you mentioned, that would be re-tracing what kotel does.
Kotel also already handles errors in its hook, hence the re-tracing too.

I've removed this span, and added a comment about it. Is that ok with you?